packages/client/src/pages/Forum/elements/SearchInput/useSearchInput.ts

```diff
@@ -0,0 +1,43 @@
+import { ChangeEvent, useCallback, useEffect, useState } from 'react'
+import { debounce } from 'lodash'
+import { useAppDispatch, useAppSelector } from '@store/hooks'
+import { forumActions, forumSelectors } from '@store/forum'
+import { getTopics } from '@store/forum/thunk'
+
+const useSearchInput = () => {
+    const searchText = useAppSelector(forumSelectors.selectSearchText)
+    const dispatch = useAppDispatch()
+    const [searchValue, setSearchValue] = useState(searchText)
+
+    const delayedDispatch = useCallback(
+        debounce((value: string) => {
+            dispatch(forumActions.setSearchText(value))
+            dispatch(getTopics({ limit: 20, search: value }))
+        }, 300),
```

Comment:
Круто, что применила `debouce`, только интервал лучше увеличить, 300мс может не хватить рядовому пользователю на поиск нужной клавиши
```suggestion
        }, 1000),
```